pkg/controller/job/controller_test.go

```diff
@@ -229,6 +231,15 @@ func TestControllerSyncJob(t *testing.T) {
 			manager.podStore.Store.Add(&pod)
 		}
 
+		// wait.ConditionFunc contract includes an error, so wrap and return nil
+		pollFn := func() (bool, error) {
+			return manager.podStoreSynced(), nil
```

Comment:
Yeah, I have the same feeling as @mikedanese, it looks as if you're delaying the problem. I'll try to have a closer look at it over the weekend, if it won't be nailed before.